app/notebook/NBSerializer.scala

```diff
@@ -227,15 +227,19 @@ object NBSerializer {
 
   implicit val notebookFormat = Json.format[Notebook]
 
-  def fromJson(json: JsValue): Notebook = {
+  def fromJson(json: JsValue): Option[Notebook] = {
     Logger.trace("\r\n**************************************\r\n")
     Logger.trace(Json.prettyPrint(json))
     Logger.trace("**************************************\r\n")
     json.validate[Notebook] match {
       case s: JsSuccess[Notebook] => {
-        val notebook: Notebook = s.get
-        val nb = notebook.cells.map { _ => notebook } getOrElse notebook.copy(cells = Some(Nil))
-        nb
+        s.get match {
+          case Notebook(None,None,None,None,None) =>
```

Comment:
Because if somebody manages to upload a file different that notebook, it still passes. Everything here: https://github.com/meh-ninja/spark-notebook/blob/master/app/notebook/NBSerializer.scala#L219-L226 is optional.